lib/libc/string/wcscmp.c

```diff
@@ -49,10 +49,9 @@ __FBSDID("$FreeBSD$");
 int
 wcscmp(const wchar_t *s1, const wchar_t *s2)
 {
-
 	while (*s1 == *s2++)
-		if (*s1++ == '\0')
+		if (*s1++ == L'\0')
 			return (0);
-	/* XXX assumes wchar_t = int */
-	return (*(const unsigned int *)s1 - *(const unsigned int *)--s2);
```

Comment:
This is worse, it's introducing signed overflow on architectures with wchar_t typedef'ed to (signed) int